tracer/src/Datadog.Trace/Tagging/TraceTagCollection.cs

```diff
@@ -220,14 +223,60 @@ public string ToPropagationHeader()
 
         public KeyValuePair<string, string>[] ToArray()
```

Comment:
We may not need `ToArray()` anymore with all the changes in this PR. If it's not called anywhere, feel free to remove it.